### PR TITLE
Feat: default layout

### PR DESCRIPTION
**Pull request by deetz99**
issue: partial bcgov/entity#22279

Done main layout, components/composables/etc except for account stuff

### CODE REVIEW

File: nuxt/layers/core/.npmignore
Comment (deetz99) [inline nuxt/layers/core/.npmignore:1]: add everything from .gitignore into .npmignore

File: nuxt/layers/core/.playground/app/app.vue
Comment (deetz99) [inline nuxt/layers/core/.playground/app/app.vue:1]: .playground not included in published layer

File: nuxt/layers/core/.playground/nuxt.config.ts
Comment (deetz99) [inline nuxt/layers/core/.playground/nuxt.config.ts:1]: .playground not included in published layer

File: nuxt/layers/core/app/app.config.ts
Comment (deetz99) [inline nuxt/layers/core/app/app.config.ts:1]: adding styles for nuxt ui components

File: nuxt/layers/core/app/app.vue
Comment (deetz99) [inline nuxt/layers/core/app/app.vue:1]: blank app.vue to use layout

File: nuxt/layers/core/app/components/Connect/BCGovLogo.vue
Comment (deetz99) [inline nuxt/layers/core/app/components/Connect/BCGovLogo.vue:1]: Only way I could get the logo alias to work when publishing the layer, will investigate better options

File: nuxt/layers/core/app/components/Connect/Header/index.vue
Comment (deetz99) [inline nuxt/layers/core/app/components/Connect/Header/index.vue:1]: base header component, will split this up when adding in account options + store

File: nuxt/layers/core/app/components/Connect/LocaleSelect.vue
Comment (deetz99) [inline nuxt/layers/core/app/components/Connect/LocaleSelect.vue:1]: locale select dropdown

File: nuxt/layers/core/app/composables/useConnectNav.ts
Comment (deetz99) [inline nuxt/layers/core/app/composables/useConnectNav.ts:1]: composable to handle navigation items and account dropdown options

File: nuxt/layers/core/app/locales/en-CA.ts
Comment (deetz99) [inline nuxt/layers/core/app/locales/en-CA.ts:1]: all english translations for current components

File: nuxt/layers/core/app/locales/fr-CA.ts
Comment (deetz99) [inline nuxt/layers/core/app/locales/fr-CA.ts:1]: all french translations for current components

File: nuxt/layers/core/app/pages/layer-dev-playground.vue
Comment (deetz99) [inline nuxt/layers/core/app/pages/layer-dev-playground.vue:1]: playground page outside of .playground, this is added to .npmignore so it wont be included when publishing layer

File: nuxt/layers/core/app/tests/unit/components/ConnectBCGovLogo.test.ts
Comment (deetz99) [inline nuxt/layers/core/app/tests/unit/components/ConnectBCGovLogo.test.ts:1]: test logo component

File: nuxt/layers/core/app/tests/unit/composables/useConnectNav.test.ts
Comment (deetz99) [inline nuxt/layers/core/app/tests/unit/composables/useConnectNav.test.ts:1]: will write tests for this once i have the account options in there properly

File: nuxt/layers/core/app/tests/unit/utils/parseSpecialChars.test.ts
Comment (deetz99) [inline nuxt/layers/core/app/tests/unit/utils/parseSpecialChars.test.ts:1]: character parser util to handle first nations/etc names. Maybe switch to package? This is not ssr friendly

File: nuxt/layers/core/app/utils/parseSpecialChars.ts
Comment (deetz99) [inline nuxt/layers/core/app/utils/parseSpecialChars.ts:1]: character parser util to handle first nations/etc names. Maybe switch to package? This is not ssr friendly

File: nuxt/layers/core/i18n.config.ts
Comment (deetz99) [inline nuxt/layers/core/i18n.config.ts:1]: required config file for i18n

File: nuxt/layers/core/nuxt.config.ts
Comment (deetz99) [inline nuxt/layers/core/nuxt.config.ts:1]: format alias and add i18n config

File: nuxt/layers/core/package.json
Comment (deetz99) [inline nuxt/layers/core/package.json:1]: use regular app instead of .playground to test layer

File: nuxt/layers/core/scripts/publish.js
Comment (deetz99) [inline nuxt/layers/core/scripts/publish.js:1]: remove typecheck for now, was working earlier but now failing on the test util imports

File: nuxt/layers/core/tailwind.config.ts
Comment (deetz99) [inline nuxt/layers/core/tailwind.config.ts:1]: lint only

File: nuxt/layers/core/app/components/Connect/Footer.vue [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          :key="link.to"
+          :to="link.to === '/' ? `/${$i18n.locale}` : link.to"
+          :target="link.target"
+          class="rounded p-1 text-sm text-white hover:underline focus:outline-none focus-visible:ring-2 focus-visible:ring-white"
+        >
+          {{ $t(link.label) }}
+        </NuxtLink>
+      </nav>
+      <span class="whitespace-nowrap italic text-bcGovColor-navDivider">{{ $t('ConnectFooter.bcApp') }}</span>
+    </div>
```
Comment (thorwolpert) [inline nuxt/layers/core/app/components/Connect/Footer.vue:50]: I added an issue to capture the version discussion
https://github.com/bcgov/entity/issues/22376

File: nuxt/layers/core/app/tests/unit/utils/parseSpecialChars.test.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  it('handles special character strings', () => {
+    const tests = [
+      ['<div>Some HTML content</div>', 'Some HTML content'],
+      ['àèìòù', 'àèìòù'],
+      ['&amp; &gt; &lt;', '& > <'],
+      ['&copy; &reg;', '© ®'],
+      ['😊', '😊'],
+      ['@#$%^&*()', '@#$%^&*()'],
+      ['&lt; &gt; &amp;', '< > &'],
+      ['&#955;UG&#695;AL&#601;S', 'λUGʷALəS'],
```
Comment (thorwolpert) [inline nuxt/layers/core/app/tests/unit/utils/parseSpecialChars.test.ts:56]: Excellent, I see you captured our FN test name.

File: nuxt/layers/core/app/utils/parseSpecialChars.ts
Comment (thorwolpert) [reply nuxt/layers/core/app/utils/parseSpecialChars.ts:1]: We can expand as we get more examples.
Link up with @kialj876 for more examples

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**